src/garvis/heartbeat_service.py

```
@@ -0,0 +1,249 @@
+            },
+        )
+
+    def run_forever(self) -> None:
+        backoff = self.interval_seconds
+        while True:
+            try:
+                self.run_once()
+                backoff = self.interval_seconds
+            except KeyboardInterrupt:
+                raise
+            except Exception:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)